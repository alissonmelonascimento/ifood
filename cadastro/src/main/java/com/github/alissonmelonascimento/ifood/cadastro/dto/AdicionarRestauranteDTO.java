package com.github.alissonmelonascimento.ifood.cadastro.dto;

import javax.validation.ConstraintValidatorContext;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.github.alissonmelonascimento.ifood.cadastro.Restaurante;
import com.github.alissonmelonascimento.ifood.cadastro.infra.DTO;

@ValidDTO
public class AdicionarRestauranteDTO implements DTO{
	
	@NotNull
    @Size(min = 3, max = 30)
	public String nomeFantasia;
    
	@NotNull
	@Pattern(regexp = "[0-9]{2}\\.[0-9]{3}\\.[0-9]{3}\\/[0-9]{4}\\-[0-9]{2}")
    public String cnpj;
    
    @NotNull
    @NotEmpty
    public String proprietario;
    
    public LocalizacaoDTO localizacao;
    
    @Override
    public boolean isValid(ConstraintValidatorContext constraintValidatorContext) {
        constraintValidatorContext.disableDefaultConstraintViolation();
        if (Restaurante.find("cnpj", cnpj).count() > 0) {
            constraintValidatorContext.buildConstraintViolationWithTemplate("CNPJ jรก cadastrado")
                    .addPropertyNode("cnpj")
                    .addConstraintViolation();
            return false;
        }
        return true;
    }

}
