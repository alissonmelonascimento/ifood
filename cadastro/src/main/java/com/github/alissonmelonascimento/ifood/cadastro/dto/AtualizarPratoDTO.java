package com.github.alissonmelonascimento.ifood.cadastro.dto;

import java.math.BigDecimal;

public class AtualizarPratoDTO {
	
	public BigDecimal preco;	

}
