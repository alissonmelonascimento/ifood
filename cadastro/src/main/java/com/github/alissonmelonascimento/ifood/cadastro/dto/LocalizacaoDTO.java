package com.github.alissonmelonascimento.ifood.cadastro.dto;

public class LocalizacaoDTO {
	
    public Double latitude;
    
    public Double longitude;	

}
