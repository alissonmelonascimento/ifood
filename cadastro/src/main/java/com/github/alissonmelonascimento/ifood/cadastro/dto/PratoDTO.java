package com.github.alissonmelonascimento.ifood.cadastro.dto;

import java.math.BigDecimal;

public class PratoDTO {
	
	public String nome;
	
	public String descricao;
	
	public BigDecimal preco;	

}
