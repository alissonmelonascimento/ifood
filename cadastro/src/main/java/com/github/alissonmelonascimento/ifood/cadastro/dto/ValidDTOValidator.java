package com.github.alissonmelonascimento.ifood.cadastro.dto;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import com.github.alissonmelonascimento.ifood.cadastro.infra.DTO;

public class ValidDTOValidator
        implements ConstraintValidator<ValidDTO, DTO> {

    @Override
    public void initialize(ValidDTO constraintAnnotation) {
    }

    @Override
    public boolean isValid(DTO dto, ConstraintValidatorContext constraintValidatorContext) {
        return dto.isValid(constraintValidatorContext);
    }
}