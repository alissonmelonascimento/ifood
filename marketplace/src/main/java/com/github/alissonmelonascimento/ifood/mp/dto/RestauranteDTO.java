package com.github.alissonmelonascimento.ifood.mp.dto;

public class RestauranteDTO {

	public String nome;
	
}
