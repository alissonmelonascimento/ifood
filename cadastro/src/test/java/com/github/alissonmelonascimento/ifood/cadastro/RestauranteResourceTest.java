package com.github.alissonmelonascimento.ifood.cadastro;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import com.github.alissonmelonascimento.ifood.cadastro.dto.AtualizarRestauranteDTO;
import com.github.alissonmelonascimento.ifood.cadastro.utils.TokenUtils;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {
	
    private String token;

    @BeforeEach
    public void gereToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
    }	

	@Test
	@DataSet("restaurantes-cenario-1.yml")//arquivo usado para inserir registros
	public void testAlterarRestaurantes() {
		
        AtualizarRestauranteDTO dto = new AtualizarRestauranteDTO();
        dto.nomeFantasia = "novoNome";
        Long parameterValue = 123L;
        given()
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()
                .statusCode(Status.NO_CONTENT.getStatusCode())
                .extract().asString();

        Restaurante findById = Restaurante.findById(parameterValue);

        //poderia testar todos os outros atribudos
        Assert.assertEquals(dto.nomeFantasia, findById.nome);
	}
	
	//Necessario para definir o content-type da chamada
	private RequestSpecification given() {
        return RestAssured.given()
                .contentType(ContentType.JSON).header(new Header("Authorization", "Bearer " + token));
	}
	
	//Exemplo PUT
	
	@Test
	@DataSet("restaurantes-cenario-1.yml")//arquivo usado para inserir registros
	public void testBuscarRestaurantes() {
		String resultado = given().when().get("/restaurantes").then().statusCode(Status.OK.getStatusCode()).extract().asString();
		
		/*
		 * Para usar isso, deve-se criar o arquivo
		 * RestauranteResourceTest.testBuscarRestaurantes.approved.json dentro da pasta
		 * desta classe com o conteudo esperado para que ele compare a saida do teste
		 * com esse arquivo e aprove ou nao o teste
		 */
		Approvals.verifyJson(resultado);
	}	
	
}
