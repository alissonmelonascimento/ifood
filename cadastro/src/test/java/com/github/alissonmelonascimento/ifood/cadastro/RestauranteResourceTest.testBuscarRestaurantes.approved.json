[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do Keyclock"
  },
  {
    "id": 124,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do Keyclock 2"
  }
]